### ISSUE TITLE
Fails to build with G729: 'audio_decoder.cpp:550:68: error: too few arguments to function ‘void bcg729Decoder(bcg729DecoderChannelContextStruct*, uint8_t*, uint8_t, uint8_t, uint8_t, uint8_t, int16_t*)’'

### ISSUE THREAD

**Issue by felixbecker2** [LubosD/twinkle#198]
Fails to build for me with `-DWITH_G729=ON` (dependency `bcg729` is installed):

```
[...]
[...]/src/audio/audio_decoder.cpp: In member function ‘virtual uint16 t_g729a_audio_decoder::decode(uint8*, uint16, int16*, uint16)’:
[...]/src/audio/audio_decoder.cpp:550:50: error: invalid conversion from ‘int16*’ {aka ‘short int*’} to ‘uint8_t’ {aka ‘unsigned char’} [-fpermissive]
  550 |   bcg729Decoder(_context, &payload[done], false, &pcm_buf[done * 8]);
      |                                                  ^~~~~~~~~~~~~~~~~~
      |                                                  |
      |                                                  int16* {aka short int*}
[...]/src/audio/audio_decoder.cpp:550:68: error: too few arguments to function ‘void bcg729Decoder(bcg729DecoderChannelContextStruct*, uint8_t*, uint8_t, uint8_t, uint8_t, uint8_t, int16_t*)’
  550 |   bcg729Decoder(_context, &payload[done], false, &pcm_buf[done * 8]);
[...]
```

(Adding `-fpermissive` to the `CFLAGS`/ `CXXFLAGS` does not remove that error.)

**Comment by choff** [comment]
Please see the pull requests #199 and #152 ; they fix this issue. However the PRs were not merged yet, unfortunately.

**Comment by fbriere** [comment]
> Please see the pull requests #199 and #152 ; they fix this issue. However the PRs were not merged yet, unfortunately.

@LubosD recently merged a bunch of PRs (thanks!), some of which were conflicting with the ones that are still left.  I'll make sure to rebase and re-submit those before long.

**Comment by fbriere** [comment]
Closing as a duplicate of #104, now fixed by #152.
